sources/api/migration/migrator/src/main.rs

```diff
@@ -82,11 +124,65 @@ fn run() -> Result<()> {
+        eprintln!("migrator is running unsigned migrations");
+        warn!("migrator is running unsigned migrations");
+        return find_and_run_unsigned_migrations(
+            &args.migration_directory,
+            &args.datastore_path, // TODO(brigmatt) make sure this is correct
+            &current_version,
+            &args.migrate_to_version,
+            &direction,
+        );
+    } else {
+        // note in the system journal that the signed code path ran.
+        info!("migrator is running signed migrations");
+        eprintln!("migrator is running signed migrations");
+    }
+    // DEPRECATED CODE END /////////////////////////////////////////////////////////////////////////
+
+    // Prepare to load the locally cached TUF repository to obtain the manifest. Part of using a
+    // `TempDir` is disabling timestamp checking, because we want an instance to still come up and
+    // run migrations regardless of the how the system time relates to what we have cached (for
+    // example if someone runs an update, then shuts down the instance for several weeks, beyond the
+    // expiration of at least the cached timestamp.json before booting it back up again). We also
+    // use a `TempDir` because see no value in keeping a datastore around. The latest  known
+    // versions of the repository metadata will always be the versions of repository metadata we
+    // have cached on the disk. More info at `ExpirationEnforcement::Unsafe` below.
+    let tough_datastore = TempDir::new().context(error::CreateToughTempDir)?;
+    let metadata_url = dir_url(&args.metadata_directory)?;
+    let migrations_url = dir_url(&args.migration_directory)?;
+
+    // Failure to load the TUF repo at the expected location is a serious issue because updog should
+    // always create a TUF repo that contains at least the manifest, even if there are no migrations.
+    let repo = tough::Repository::load(
+        &tough::FilesystemTransport,
+        tough::Settings {
+            root: File::open(&args.root_path).context(error::OpenRoot {
+                path: args.root_path.clone(),
+            })?,
+            datastore: tough_datastore.path(),
+            metadata_base_url: metadata_url.as_str(),
+            targets_base_url: migrations_url.as_str(),
+            limits: REPOSITORY_LIMITS,
+            // The threats TUF mitigates are more than the threats we are attempting to mitigate
+            // here by caching signatures for migrations locally and using them after a reboot but
+            // prior to Internet connectivity. We are caching the TUF repo and use it while offline
+            // after a reboot to mitigate binaries being added or modified in the migrations
+            // directory; the TUF repo is simply a code signing method we already have in place,
+            // even if it's not one that initially makes sense for this use case.
+            expiration_enforcement: ExpirationEnforcement::Unsafe,
```

Comment:
```suggestion
            // even if it's not one that initially makes sense for this use case.
            // So, we don't care if the targets expired between updog downloading them and now.
            expiration_enforcement: ExpirationEnforcement::Unsafe,
```
I feel like the explanation was good but didn't actually say why / that we don't care about expiration :)

Reply:
Addressed in [4f34544](https://github.com/bottlerocket-os/bottlerocket/compare/afc4b5d9ddfb7c6fb24f30abdaf6a5a000df222c..4f345446016e314c3da3f0855c92db5ca979f2a4)